lib/collection/src/collection/state_management.rs

```diff
@@ -29,12 +30,22 @@ impl Collection {
         this_peer_id: PeerId,
         abort_transfer: impl FnMut(ShardTransfer),
     ) -> CollectionResult<()> {
-        self.apply_config(state.config).await?;
-        self.apply_shard_transfers(state.transfers, this_peer_id, abort_transfer)
+        let State {
+            config,
+            shards,
+            resharding,
```

Comment:
We totally missed handling `resharding` here. Struct destruction reveals the unused field.